packages/shared/src/components/auth/AuthModalFooter.tsx

```diff
@@ -1,33 +1,35 @@
-import React, { ReactElement } from 'react';
+import React, { MouseEventHandler, ReactElement } from 'react';
 import classNames from 'classnames';
 import { ClickableText } from '../buttons/ClickableText';
 import { Modal } from '../modals/common/Modal';
 import { Justify } from '../utilities';
 
+interface TextType {
+  body?: string;
+  button: string;
+}
 interface AuthModalFooterProps {
   className?: string;
-  isLogin: boolean;
-  onIsLogin: (value: boolean) => void;
+  text: TextType;
+  onClick: MouseEventHandler;
 }
 
 function AuthModalFooter({
   className,
-  isLogin,
-  onIsLogin,
+  text,
+  onClick,
 }: AuthModalFooterProps): ReactElement {
   return (
     <Modal.Footer
       className={classNames(className, 'gap-unset')}
       justify={Justify.Center}
     >
-      <Modal.Text>
-        {isLogin ? 'Don’t have an account?' : 'Already have an account?'}
-      </Modal.Text>
+      {text.body && <Modal.Text>{text.body}</Modal.Text>}
       <ClickableText
         className="ml-1 underline text-theme-label-primary"
-        onClick={() => onIsLogin(!isLogin)}
+        onClick={onClick}
       >
-        {isLogin ? 'Sign up' : 'Log in'}
+        {text.button}
```

Comment:
this looks funky, what is the difference between `text.body` and `text.button` here?

Reply:
We render a button which has some text in it, and some descriptive optional text on top.